abi-versions/v0.2.1/README.md

```diff
@@ -1791,6 +1908,66 @@ Returned `errno` value is:
 This function is never called.
 
 
+# Serialization
+
+> **Note**
+> The encoding of integers is little-endian.
+
+
+#### Maps with HTTP fields and/or gRPC metadata
+
+Non-empty maps are serialized as:
+- 32-bit integer containing the number of keys in the map,
+- a series of pairs of 32-bit integers containing length of key and value,
+- a series of pairs of bytes containing key and value, separated by `NULL`
```

Comment:
nit: "key and value, separated by `NULL` character" could be misinterpreted as meaning that there is a null char between key and value (only), but not between the pair itself and the next pair.

How about something like, "with each key and value byte sequence terminated by a `NULL` (`0x00`) character"?

Reply:
Might also be good to explicitly state that the terminating NULL characters are not counted in the key/value length values written earlier (even though I think the current language already implies that, strictly speaking).

Reply:
Good point. Done. Thanks!